src/Http/Actions/Author/AuthorGetProfileAction.php

```diff
@@ -128,13 +128,19 @@ class AuthorGetProfileAction extends Action
         )
     ]
     public function __invoke(
-        int $authorId,
+        string|int $author,
         Request $request
     ): BaseHttpResponse|JsonResponse|JsonResource|RedirectResponse {
+        /** @var User $user */
         $user = User::query()
-            ->with('posts')
-            ->whereId($authorId)
-            ->first();
+            ->with('posts');
+        if (is_numeric($author)) {
```

Comment:
- if:
(int) value  > 0 => get by ID
    => if user not found 
    => find by username
- else find by username